src/ReleaseNotesUtil/GenDiffNotes.cmd

```diff
@@ -0,0 +1,41 @@
+@echo off
+REM GenDiffNotes.cmd scratchDir nugetSource fxCopAnalyzersOldVersion fxCopAnalyzersNewVersion
+REM Ex: GenDiffNotes.cmd C:\scratch nuget.org 2.6.3 2.9.0
+
+if [%1]==[] goto usage
+if [%2]==[] goto usage
+if [%3]==[] goto usage
+if [%4]==[] goto usage
+
+set SCRATCHDIR=%1
+set NUGETSOURCE=%2
+set OLDVERSION=%3
+set NEWVERSION=%4
+
+set DIFFNOTES=%SCRATCHDIR%\%OLDVERSION%_%NEWVERSION%-notes.md
+
+@echo on
+mkdir %SCRATCHDIR%
+pushd %SCRATCHDIR%
+nuget.exe install -source %NUGETSOURCE% Microsoft.CodeAnalysis.FxCopAnalyzers -version %OLDVERSION%
+nuget.exe install -source %NUGETSOURCE% Microsoft.CodeAnalysis.FxCopAnalyzers -version %NEWVERSION%
+popd
+dotnet.exe ReleaseNotesUtil.dll getrulesjson %SCRATCHDIR% %OLDVERSION% %SCRATCHDIR%\%OLDVERSION%-rules.json
```

Comment:
I am wondering if we need such a complicated tooling. We already have tooling to [generate .md files](https://github.com/dotnet/roslyn-analyzers/blob/master/src/GenerateAnalyzerRulesets/Program.cs#L374-L416), [.ruleset files](https://github.com/dotnet/roslyn-analyzers/blob/master/src/GenerateAnalyzerRulesets/Program.cs#L137-L275) and [.sarif files](https://github.com/dotnet/roslyn-analyzers/blob/master/src/GenerateAnalyzerRulesets/Program.cs#L419-L545) that contains the categorized rule information for all rules implemented in the package. I am wondering if we can update that tooling to auto-generate the rule diff info and add it to the NuGet package alongside other packaged documentation files. We would then just copy the diff file into the release notes.

Note my primary concern is from maintenance of this new tooling, but if you feel it won't be an overhead, then I am fine with the new tooling approach.

Reply:
Part of my ulterior motive for making this separate from the existing GenerateAnalyzerRulesets is making it easier to steal for other purposes.

I agree it would be nice to have everything automated.  We'd need to maintain "previous" version somewhere to diff against.  I dunno when I would have time to do something like that, though.  Let's see how painful it is to run manually every release?

I also wanted to make it easy to generate diffs for previous release notes.

Reply:
Sounds reasonable. We can always start with your current tool, and if it becomes difficult to maintain, think about alternate route.